File: zeromq/cpp/hello_world_server.cpp
#include <zmq.hpp>
#include <string>
#include <iostream>
#include <unistd.h>

int main(){
    zmq::context_t context(1);
    zmq::socket_t socket (context, ZMQ_REP);
    socket.bind("tcp://*:5555");
    while (true){
        zmq::message_t request;
        socket.recv(&request);
        std::string *req_str = static_cast<std::string *>(request.data());
        std::cout << "Received request: " <<  *req_str << std::endl;
        sleep(1);

        zmq::message_t reply(5);
        memcpy(reply.data(), "World", 5);
        socket.send(reply);
    }
    return 0;
}

File: zeromq/java/zmq-demo/src/main/java/com/saihei/demo/HelloWorldServer.java
package com.saihei.demo;
import org.zeromq.ZMQ;

public class HelloWorldServer{
    public static void main(String[] args) throws Exception{
        ZMQ.Context context = ZMQ.context(1);
        ZMQ.Socket socket = context.socket(ZMQ.REP);
        socket.bind("tcp://localhost:5555");
        while (!Thread.currentThread().isInterrupted()){
            byte[] request = socket.recv(0);
            System.out.println("Received : [" + new String(request, ZMQ.CHARSET) + "]");
            String reply = "World";
            socket.send(reply.getBytes(ZMQ.CHARSET), 0);
            Thread.sleep(1000);
        }
        socket.close();
        context.term();
    }
}
